### PR TITLE
Improves clarity in Sphinx documentation for function signature.

### LINKED ISSUE #11695: pytest.exit returncode default is poorly documented
According to <https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-exit>, the default value of the `returncode` parameter of `pytest.exit` is `False`. However:

* According to the actual source, the default value is `None`, not `False`: <https://docs.pytest.org/en/7.1.x/_modules/_pytest/outcomes.html#exit>.

* It's not clear what a `returncode` of `None` actually means. You can't return `None` to the OS (the return code is an integer), so it must correspond to some integer value.

### PR DESCRIPTION
**Pull request by arthuRHD**
closes #11695.

### CODE REVIEW

File: src/_pytest/outcomes.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def exit(
 
     :param returncode:
         Return code to be used when exiting pytest.
+    :default returncode: None
```
Comment (nicoddemus) [inline src/_pytest/outcomes.py:116]: IIUC, the problem was not that it was unclear that `returncode` might be `None`, the source of confusion was "what None *means*", so I think this should instead be something like:

```suggestion
    :param returncode:
        Return code to be used when exiting pytest. None means the same as ``0`` (no error), same as :func:`sys.exit`.
```

**Review by nicoddemus — APPROVED** [review]
Thanks @arthuRHD!

**Comment by bluetech** [comment]
To be precise, `None` does not mean the same as `0`:

- Setting `returncode=0` means "exit with 0" (i.e. override the exit code to 0).
- Setting `returncode=None` means "exit with `session.exitstatus`" (i.e. don't override). Normally `session.exitstatus` is 0 but it can be set to something else by plugins. Even though `session.exitstatus` is undocumented some plugins do set it.

**Comment by arthuRHD** [comment]
> Thanks @arthuRHD! 

You guys did everything, thanks

**Pull request merged**